src/remote-backend/find-zone.cc

```diff
@@ -26,9 +26,17 @@ string RemoteBackend::findBestZone(const string& qname) {
   // I miss dnsname.cc :(
   std::vector<std::string> labels;
   boost::split(labels, qname, boost::is_any_of("."));
-  string ret = qname;
   auto session = getSession();
 
+  if (labels.at(0) == "*") {
+    labels.erase(labels.begin());
+  }
+
+  string ret = boost::algorithm::join(labels, ".");
+  if (ret.empty()) {
+    ret = ".";
```

Comment:
```suggestion
    return ".";
```

Reply:
No, as we need to throw an `std::out_of_range` to indicate there is no zone for the qname.

Reply:
right!